src/site/antora/modules/ROOT/pages/manual/json-template-layout.adoc

```diff
@@ -1085,7 +1085,7 @@ Resolves `logEvent.getMessage().getParameters()`.
 Regarding garbage footprint, `stringified` flag translates to
 `String.valueOf(value)`, hence mind not-`String`-typed values. Further,
 `logEvent.getMessage()` is expected to implement `ParameterVisitable` interface,
-which is the case if `log4j2.enableThreadLocals` property set to true.
+which is the case if `log4j2.enableThreadlocals` property set to true.
```

Comment:
I'd be in favor of using `enableThreadLocals` (just like many other camelcase property names we have) and consistently using it everywhere in the code.

Reply:
Me, too, except backward compatibility prevents me from doing it. :wink:

The original property is `log4j2.enable.threadlocals`, which has 2 tokens `enable` and `threadlocals`. Therefore the normalized form is `log4j2.enableThreadlocals`.

Reply:
The matching is not so fuzzy. For a given legacy property name (`log4j2.enable.threadlocals`), `PropertiesUtil` looks for:

* a property of the same name,
* a property with a normalized name (`LOG4J_ENABLE_THREADLOCALS` for environment variables, `log4j2.enableThreadlocals` for all other property sources),
* a property that contains the same tokens (e.g. `org.apache.logging.log4j.enableThreadlocals`). The tokenizer is in `PropertySource.Util`.